tests/console/docker_image.pm

```diff
@@ -7,69 +7,70 @@
-
-        # For Tumbleweed, the image is wrapped inside an RPM
-        zypper_call "in docker opensuse-tumbleweed-image";
+    if (script_run("SUSEConnect --status-text") != 0) {
+        assert_script_run("SUSEConnect --cleanup");
+        assert_script_run("SUSEConnect -r $SCC_REGCODE");
+        add_suseconnect_product("sle-module-containers", substr($version, 0, 2));
     }
-    elsif (is_leap('15.0+')) {
-        $image_name = "opensuse-leap-$version:current";
-        $image_path = "/usr/share/suse-docker-images/native/opensuse-leap${version}-image.tar.xz";
 
-        # For Leap, the docker image is the ISO asset
-        my $image_filename = get_required_var('ISO');
-        $image_filename =~ s/^.*\///;
-        my $image_url = autoinst_url("/assets/other/$image_filename");
-        assert_script_run "curl $image_url --create-dirs -o $image_path";
+    my $image_name;
+    if (is_sle("=12-SP3")) {
+        $image_name = "registry.suse.de/suse/sle-12-sp3/update/products/casp30/container/sles12:sp3";
+    }
+    elsif (is_sle("=15")) {
+        $image_name = "registry.suse.de/suse/sle-15/update/cr/images/suse/sle15:current";
     }
     else {
-        die 'Only know about Tumbleweed and Leap 15.0+ docker images';
+        die("This test only works at SLE12SP3 and SLE15.");
     }
 
-    # Start the docker daemon, normally done by previous test modules already
-    systemctl 'start docker';
-    systemctl 'status docker';
-    assert_script_run 'docker info';
+    # Allow our internal 'insecure' registry
+    zypper_call("in docker");
+    assert_script_run("mkdir -p /etc/docker");
+    assert_script_run('cat /etc/docker/daemon.json; true');
+    assert_script_run(
+        'echo "{ \"insecure-registries\" : [\"registry.suse.de\", \"registry.suse.de:443\", \"registry.suse.de:5000\"] }" > /etc/docker/daemon.json');
+    assert_script_run('cat /etc/docker/daemon.json');
+    systemctl('restart docker');
 
     # Load the image
-    assert_script_run "docker load -i $image_path";
-    # Show that the image got registered
-    assert_script_run "docker images $image_name";
+    assert_script_run("docker pull $image_name", 120);
     # Running executables works
-    assert_script_run qq{docker container run --rm $image_name echo "I work" | grep "I work"};
-    # It is the correct openSUSE version
-    validate_script_output qq{docker container run --rm $image_name cat /etc/os-release}, sub { /PRETTY_NAME="openSUSE (Leap )?${version}( .*)?"/ };
-    # Zypper and network are working
-    assert_script_run qq{docker container run --rm $image_name zypper -v ref | grep "All repositories have been refreshed"};
+    assert_script_run("docker container run --rm $image_name echo 'I work' | grep 'I work'");
+    # It is the right SLE version
+    #   validate_script_output qq{docker container run --rm $image_name cat /etc/os-release}, sub { /PRETTY_NAME="SUSE Linux Enterprise Server 15"/ };
```

Comment:
Should this SLE version be a real check ? if it's a SLE 15 test then PRETTY_NAME should be ...15...

Reply:
:+1: